src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProvider.cs

```diff
@@ -251,24 +243,16 @@ public override bool TryGetScaleMonitor(
         {
             lock (this.initLock)
             {
-                if (this.singletonScaleMonitor == null)
+                if (this.singletonDurableTaskMetricsProvider == null)
                 {
-                    DurableTaskMetricsProvider metricsProvider = this.GetMetricsProvider(
+                    // This is only called by the ScaleController, it doesn't run in the Functions Host process.
+                    this.singletonDurableTaskMetricsProvider = this.GetMetricsProvider(
                         hubName,
                         this.clientProviderFactory.GetClientProvider(connectionName),
                         this.logger);
-
-                    // Scalers in Durable Functions are shared for all functions in the same task hub.
-                    // So instead of using a function ID, we use the task hub name as the basis for the descriptor ID.
-                    string id = GetScalerUniqueId(hubName);
-                    this.singletonScaleMonitor = new DurableTaskScaleMonitor(
-                        id,
-                        hubName,
-                        this.logger,
-                        metricsProvider);
                 }
 
-                scaleMonitor = this.singletonScaleMonitor;
+                scaleMonitor = new DurableTaskScaleMonitor(functionId, hubName, this.logger, this.singletonDurableTaskMetricsProvider);
```

Comment:
This constructor currently has the following logic:
```csharp
// Scalers in Durable Functions are shared for all functions in the same task hub.
// So instead of using a function ID, we use the task hub name as the basis for the descriptor ID.
this.scaleMonitorDescriptor = new ScaleMonitorDescriptor(id: id, functionId: id);
```
Is this use of the ID for both the `id` and `functionId` parameters okay or do we need to update that as well?

Reply:
Made change for the line to be consistent with:

https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Scale/ScaleMonitorDescriptor.cs#L32

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScaleMonitor.cs#L38